src/media-query-provider.js

```diff
@@ -1,68 +1,87 @@
 import React from 'react';
-import matchMedia from 'matchmedia';
 import PropTypes from 'prop-types';
 import shallowequal from 'shallowequal';
-
-const defaultQueries = {
-  mobile: 'screen and (max-width: 623px)',
-  tablet: 'screen and (min-width: 624px) and (max-width: 1020px)',
-  desktop: 'screen and (min-width: 1021px) and (max-width: 1440px)',
-  largeDesktop: 'screen and (min-width: 1441px)',
-};
+// this is for server side rendering and does not use window.matchMedia
+import cssMediaQuery from 'css-mediaquery';
+import isNode from 'detect-node';
 
 class MediaQueryProvider extends React.Component {
-
   constructor(props) {
     super(props);
+
+    const queryTuples = Object.entries(this.props.queries);
+
+    const media = queryTuples.reduce((acc, [queryName, query]) => {
+      if (this.props.values) {
+        acc[queryName] = cssMediaQuery.match(query, this.props.values);
+      } else {
+        // if the consumer has not set `values` and is server rendering, default to false
+        // because we don't know the screen size
+        acc[queryName] = isNode ? false : window.matchMedia(query).matches;
```

Comment:
as mentioned use `domExists` func i supplied in comments to avoid the extra library